barretenberg/cpp/src/barretenberg/ultra_honk/oink_prover.cpp

```diff
@@ -149,6 +153,15 @@ template <IsUltraFlavor Flavor> void OinkProver<Flavor>::execute_grand_product_c
     transcript->send_to_verifier(domain_separator + commitment_labels.z_lookup, witness_commitments.z_lookup);
 }
 
+template <IsUltraFlavor Flavor> typename Flavor::RelationSeparator OinkProver<Flavor>::generate_alphas_round()
+{
+    RelationSeparator alphas;
+    for (size_t idx = 0; idx < alphas.size(); idx++) {
+        alphas[idx] = transcript->template get_challenge<FF>(domain_separator + "alpha_" + std::to_string(idx));
```

Comment:
Just to make sure I understand, the transcript will now generate arbitrarily many new challenges from the same underlying data? Its adding some additional content internally or something?

Reply:
the transcript will generate a challenge based on the previous challenge, but there won't be any other new content when we generate challenges consecutively